go.mod

```diff
@@ -3,6 +3,13 @@ module github.com/notaryproject/notation-go
 go 1.17
 
 require (
+	github.com/go-ldap/ldap/v3 v3.4.3
```

Comment:
`go-ldap` has a dependency on `github.com/Azure/go-ntlmssp`, but the functionality we need, [DN validation](github.com/Azure/go-ntlmssp) does not. Is there a way to avoid this transitive dependency? Is the transitive dependency dropped during build as we do not use any types from `github.com/Azure/go-ntlmssp`?

Reply:
AFAIK, final build will only include the specific methods and their dependencies. Although I could not confirm this with official Go docs, I found multiple online forums confirming this. Seems like https://github.com/jondot/goweight could help us see what's included in the final build. I added it to my TODO list to revisit later.

Reply:
unpopular opinion: Pulling in `github.com/go-ldap/ldap`, which is a library to talk with LDAP servers, just for validating a DN, seems like using a sledgehammer to crack a nut.

I would argue it's better just to copy `ldap.ParseDN`, and related tests, into `notation-go` and remove that dependency.

Reply:
@qmuntal If we consume the whole package in our builds (like in java), then I agree with what you said. Thanks to Go static linking, our build will only include `dn.ParseDN` and it's dependencies. I prefer to keep a dependency on `go-ldap` to ensure we get the updates and security fixes. Also, `go-ldap` is the most popular library for ldap in Go, I'm not too worried about having a dependency on it.

Reply:
Besides, I want to give them the credit rather than simply copying their code :)

Reply:
I'm not that concerned about binary size, as Go is really good at prunning unnecessary code. I'm worried about having a sane dependency tree. Keeping it to the barr minimum is always a plus, even if it means duplicating code.

That said, this particular dpeendency is not a red line for me.

Reply:
Tracking this in https://github.com/notaryproject/notation-go/issues/45